bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -995,7 +1124,37 @@ public void putResource(Resource resource) throws ApiException, InterruptedExcep
                 throw new ApiException("Parsing error", e);
             }
         } catch (ExecutionException | TimeoutException e) {
-            throw new ApiException("putResource() error sending request", e);
+            throw new ApiException("Error sending PUT request", e);
+        } finally {
+            closeStream(stream);
+        }
+    }
+
+    /**
+     * Close and re-open the session. Called when the server sends a GO_AWAY message. Acquires a SessionSynchronizer
+     * 'write' lock to ensure single thread access while the new session is being created. Therefore it waits for any
+     * already running GET/PUT method calls, which have a 'read' lock, to complete. And also causes any new GET/PUT
+     * method calls to wait until this method releases the 'write' lock again. Whereby such GET/PUT calls are postponed
+     * to the new session.
+     */
+    private synchronized void recycleSession() {
```

Comment:
Is 'recycle' the right wording here given the session is actually thrown away? How about `recreateSession()`?

Reply:
Ok.